Assets/MRTK/SDK/Experimental/Features/UX/BoundsControl/Visuals/TranslationHandles.cs

```diff
@@ -0,0 +1,338 @@
+            }
+
+            // Even handle indices point in the positive direction along their axis,
+            // and odd handle indices point in the negative direction;
+            var directionSign = handleIndex % 2 == 0 ? 1.0f : -1.0f;
+
+            // Align handle with its edge assuming that the prefab is initially aligned with the up direction 
+            if (faceAxes[handleIndex] == CardinalAxisType.X)
+            {
+                Quaternion realignment = Quaternion.FromToRotation(Vector3.forward, directionSign * Vector3.right);
+                midpointVisual.transform.localRotation = realignment * midpointVisual.transform.localRotation;
+            }
+            else if (faceAxes[handleIndex] == CardinalAxisType.Z)
+            {
+                Quaternion realignment = Quaternion.FromToRotation(Vector3.forward, directionSign * Vector3.forward);
+                midpointVisual.transform.localRotation = realignment * midpointVisual.transform.localRotation;
+            } else if (faceAxes[handleIndex] == CardinalAxisType.Y)
+            {
+                Quaternion realignment = Quaternion.FromToRotation(Vector3.forward, directionSign * Vector3.up);
+                midpointVisual.transform.localRotation = realignment * midpointVisual.transform.localRotation;
+            }
+
+            Bounds midpointBounds = VisualUtils.GetMaxBounds(midpointVisual);
+            float maxDim = VisualUtils.GetMaxComponent(midpointBounds.size);
+            float invScale = maxDim == 0.0f ? 0.0f : config.HandleSize / maxDim;
+
+            midpointVisual.name = "visuals";
+            midpointVisual.transform.parent = parent.transform;
+            midpointVisual.transform.localScale = new Vector3(invScale, invScale, invScale);
+            midpointVisual.transform.localPosition = Vector3.zero;
+
+            if (config.HandleMaterial != null)
+            {
+                VisualUtils.ApplyMaterialToAllRenderers(midpointVisual, config.HandleMaterial);
+            }
+
+            return midpointBounds;
+        }
+
+        #region BoundsControlHandlerBase 
+        internal override bool IsVisible(Transform handle)
+        {
+            CardinalAxisType axisType = GetAxisType(handle);
+            return IsActive &&
+                ((axisType == CardinalAxisType.X && config.ShowTranslationHandleForX) ||
+                (axisType == CardinalAxisType.Y && config.ShowTranslationHandleForY) ||
+                (axisType == CardinalAxisType.Z && config.ShowTranslationHandleForZ));
+        }
+
+        internal override HandleType GetHandleType()
+        {
+            return HandleType.Translation;
+        }
+
+        protected override Transform GetVisual(Transform handle)
+        {
+            // visual is first child 
+            Transform childTransform = handle.GetChild(0);
+            if (childTransform != null && childTransform.name == "visuals")
```

Comment:
Same comment for "visuals" here and using const string.

Reply:
@thalbern Deferring to you here; using string compare for handle management